app/dashboard/src/pages/authentication/Login.tsx

```diff
@@ -33,119 +33,92 @@ import TextLink from '#/components/TextLink'
 /** A form for users to log in. */
 export default function Login() {
   const location = router.useLocation()
-  const { signInWithGoogle, signInWithGitHub, signInWithPassword } = authProvider.useAuth()
-  const { getText } = textProvider.useText()
-
+  const auth = useAuth()
+  const { getText } = useText()
   const query = new URLSearchParams(location.search)
   const initialEmail = query.get('email')
-
-  const [email, setEmail] = React.useState(initialEmail ?? '')
-  const [password, setPassword] = React.useState('')
-  const [isSubmitting, setIsSubmitting] = React.useState(false)
-  const shouldReportValidityRef = React.useRef(true)
-  const formRef = React.useRef<HTMLFormElement>(null)
-
-  const localBackend = backendProvider.useLocalBackend()
+  const localBackend = useLocalBackend()
   const supportsOffline = localBackend != null
 
+  const signInWithGoogleMutation = useMutation({ mutationFn: auth.signInWithGoogle })
```

Comment:
Not sure if we need additional wrappers here, or we can pull them up to AuthProvider?

Reply:
i mean, do we need mutations at all?
if we do want a mutation for everything then... maybe we could imperatively get the form values to avoid sending them as variables, i guess that's not quite ideal though...